### PR TITLE
feat: add usage explorer mv

### PR DESCRIPTION
**Pull request by makeavish**

### CODE REVIEW

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE IF NOT EXISTS signoz_traces.usage_explorer (
+  timestamp DateTime64(9) CODEC(DoubleDelta, LZ4),
+  service_name LowCardinality(String) CODEC(ZSTD(1)),
+  count UInt64 CODEC(T64, ZSTD(1))
+) ENGINE SummingMergeTree()
+PARTITION BY toDate(timestamp)
+ORDER BY (timestamp, service_name);
```
Comment (ankitnayan) [inline exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql:7]: @makeavish high cardinality sorting key before low cardinality is not recommended IMO

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS signoz_traces.usage_explorer_mv
+TO signoz_traces.usage_explorer
+AS SELECT
+  toStartOfHour(timestamp) as timestamp,
+  serviceName as service_name,
+  count() as count
+FROM signoz_traces.signoz_index_v2
+GROUP BY toStartOfHour(timestamp) as timestamp, serviceName;
```
Comment (ankitnayan) [inline exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql]: should this be just `GROUP BY timestamp, serviceName`

**Review by ankitnayan — CHANGES_REQUESTED** [review]

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql (hunk 1 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql:7]: But in all of the cases we are using timestamp as filter and service filter is secondary filter.

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql (hunk 2 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql]: Updated

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql (hunk 1 above)
Comment (ankitnayan) [reply exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.up.sql:7]: okay..then this should be fine. A query with `service_name` and without `timestamp` would perform poor in the current schema

**Review by ankitnayan — APPROVED** [review]

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.down.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+DROP TABLE IF EXISTS signoz_traces.usage_explorer
+DROP VIEW IF EXISTS signoz_traces.usage_explorer_mv
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.down.sql:2]:
```suggestion
DROP TABLE IF EXISTS signoz_traces.usage_explorer;
DROP VIEW IF EXISTS signoz_traces.usage_explorer_mv;
```

Never tried migrate down. Are semicolons required?

**Review by srikanthccv — APPROVED** [review]

File: exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.down.sql (hunk 3 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.down.sql:2]: Thanks for pointing out, down is failing even before usage explorer.
Creating a separate PR for the same.

Comment (makeavish) [reply exporter/clickhousetracesexporter/migrations/000014_add_usage_explorer_mv.down.sql:2]: Please review this PR: https://github.com/SigNoz/opentelemetry-collector-contrib/pull/2074

**Pull request merged**